lib/config/generator.js

```diff
@@ -0,0 +1,222 @@
+var path = require('path');
+var Vow = require('vow');
+var vowNode = require('vow-node');
+var Table = require('cli-table');
+var prompt = require('prompt');
+var fs = require('fs');
+
+var Checker = require('../checker');
+var utils = require('../utils');
+
+prompt.message = '';
+prompt.delimiter = '';
+prompt.start();
+
+/**
+ * Script that walks the user through the autoconfig flow
+ * @type {String[]}
+ */
+var prompts = [
+    {
+        name: 'Please choose a preset number:'.green,
+        require: true,
+        pattern: /[0-9]*/
+    },
+    {
+        name: 'Create an (e)xception for this rule, or (f)ix the errors yourself?'.green,
+        require: true,
+        pattern: /(e|f)/
+    }
+];
+
+function Generator() {
+    this._config = {};
+}
+
+Generator.prototype = {
+    /**
+     * Generates a configuration object based for the given path
+     * based on the best fitting preset
+     *
+     * @param  {String} path - The path containing file(s) used to guide the configuration
+     *
+     * @return {Promise} Resolved with the generated, JSCS configuration
+     */
+    generate: function(path) {
+        var checker = getChecker();
+        var _path = normalizePath(path, checker.getBasePath());
+        var presetNames = Object.keys(checker.getPresets());
+        var _this = this;
+        var errorsCollection;
+
+        console.log('Checking', _path, 'against the presets');
+
+        return Vow
+        .all(presetNames.map(this._checkAgainstPreset.bind(this, _path)))
+        .then(function generateErrorsCollection(results) {
+            errorsCollection = [].concat.apply([], results);
+            return errorsCollection;
+        })
+        .then(this._showErrorCounts.bind(this, presetNames))
+        .then(this._getUserPresetChoice.bind(this, prompts[0]))
+        .then(function getPresetIndexFromPrompt(choiceObj) {
```

Comment:
Combine this and the next two thens. If you want to use functions for clarity, go ahead, but no need for multiple thens, it's just noisy.